src/main/java/qwerty/Parser.java

```diff
@@ -0,0 +1,85 @@
+package qwerty;
+
+import java.util.HashMap;
+import java.util.Scanner;
+
+import qwerty.command.ByeCommand;
+import qwerty.command.Command;
+import qwerty.command.DeadlineCommand;
+import qwerty.command.DeleteCommand;
+import qwerty.command.EmptyCommand;
+import qwerty.command.EventCommand;
+import qwerty.command.FindCommand;
+import qwerty.command.ListCommand;
+import qwerty.command.MarkCommand;
+import qwerty.command.TodoCommand;
+import qwerty.command.UnknownCommand;
+import qwerty.command.UnmarkCommand;
+
+/**
+ * This class encapsulates a Parser that parses command line inputs from the user.
+ * The parser receives raw string inputs and returns the corresponding Command object
+ * created with the given input arguments.
+ */
+public class Parser {
+    /**
+     * Returns a map of parameter-argument pairs as strings.
+     * The map is generated from parsing the argument string after the main command.
+     *
+     * @param additionalArgs String containing arguments input from the user.
+     * @return A hashmap containing parameters and their arguments
+     */
+    public static HashMap<String, String> parseArgs(String additionalArgs) {
+        HashMap<String, String> map = new HashMap<>();
+
+        // extract first argument separately because it has no identifier
+        String[] argSplit = additionalArgs.split(" /", 2);
```

Comment:
Perhaps could be named `argSplits` to reflect the plural nature of the variable?